enhancements/oc/oc-login-external-oidc-integration.md

```diff
@@ -0,0 +1,228 @@
+
+oc login generates a stanza in kubeconfig supporting credentials exec plugin feature in client-go.
+
+For example;
+
+```shell
+$ oc login localhost:8443 ---issuer-url=https://oidc.issuer.url -exec-plugin=oc-oidc --client-id=client-id --extra-scopes=email,profile --callback-port=8080
+```
+
+results in;
+
+```yaml
+- name: localhost:8443
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1
+      args:
+      - get-token
+      - --issuer-url=https://oidc.issuer.url
+      - --client-id=client-id
+      - --callback-address=127.0.0.1:8080
+      - --extra-scopes=email,profile
+      command: oc
+      env: null
+      installHint: Please be sure that oc is defined in $PATH to be executed as credentials
+        exec plugin
+      interactiveMode: IfAvailable
+      provideClusterInfo: false
+```
+
+Moreover, `oc login` command triggers authentication process by sending a request to Project API's whoami endpoint automatically so that
+client-go talks to `oc get-token` command to obtain id token (and also refresh token, if provider supports it).
+
+There are currently 7 authentication flows grouped around public and confidential clients. Due to the nature of `oc` that is being used widely on local, 
+we have decided to support authentication flows only for public clients for now but if there is any different use case in the future, this command can support confidential clients too. 
+
+Supportability (in terms of authentication flows) matrix of this feature is outlined below; 
+
+|           | Authorization Code<br/>(public/confidential)                        | Authorization Code + PKCE<br/>(public/confidential)                 | Implicit Flow<br/>(public) | Client Credentials<br/>(confidential)                                    | Refresh Token<br/>(public/confidential) | Device Code<br/>(public) | Password Grant<br/>(confidential) |
+|-----------|---------------------------------------------------------------------|---------------------------------------------------------------------|----------------------------|--------------------------------------------------------------------------|-----------------------------------------|--------------------------|-----------------------------------|
+| Supported | Yes                                                                 | Yes                                                                 | No                         | No                                                                       | Yes                                     | No                       | No                                |
+| Comment   | Can be used for confidential clients<br/>by passing --client-secret | Can be used for confidential clients<br/>by passing --client-secret | Deprecated                 | Mostly used in CI systems that <br/>client-secret can be stored securely | If provider supports it                 |                          | Deprecated                        |
```

Comment:
Indicate who deprecated it.  We need to be clear whether it was our choice or a standard choice.

Reply:
Added.